src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -63,12 +63,136 @@ private BoundExpression BindIsPatternExpression(IsPatternExpressionSyntax node,
             DiagnosticBag diagnostics)
         {
             var type = (NamedTypeSymbol)this.BindType(node.Type, diagnostics);
+
+            var operators = type.GetMembers(WellKnownMemberNames.IsOperatorName);
```

Comment:
It doesn't make sense for `operator is`. If you're asking if it "is" the derived type, we should not answer yes because it "is" the base type.

Your concern is noted, but this is the current shape of the feature.